core/src/main/java/ai/timefold/solver/core/api/score/stream/ConstraintFactory.java

```diff
@@ -65,6 +67,14 @@ public interface ConstraintFactory {
      */
     <A> @NonNull UniConstraintStream<A> forEach(@NonNull Class<A> sourceClass);
 
+    /**
+     * As defined by {@link #forEach(Class)},
+     * but including filtered entities that match the supplied {@link ForEachInclude includes}.
+     *
+     * @param <A> the type of the matched problem fact or {@link PlanningEntity planning entity}
+     */
+    <A> @NonNull UniConstraintStream<A> forEachIncluding(@NonNull Class<A> sourceClass, ForEachInclude... includes);
```

Comment:
So, the code would look like this:

    forEachIncluding(Visit.class, UNASSIGNED, INCONSISTENT);

I have to say I still prefer the following:

    forEach(Visit.class, INCLUDE_UNASSIGNED, INCLUDE_INCONSISTENT);

This way, it's crystal clear that the includes are "AND" (because they both have the "INCLUDE_" prefix). The "Including" suffix in the name of the method is IMO not sufficient for that, because there is no obvious connection between the name and the method and the enum values.

Yes, the user could read the Javadoc, and arguably they should.
But arguably, code which can be understood even without reading any documentation is better than code which does not.